components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java

```diff
@@ -0,0 +1,214 @@
+
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.common.testng.WithRealmService;
+import org.wso2.carbon.identity.common.testng.WithRegistry;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.dto.PublisherDataHolder;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementConfigHolder;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent;
+
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+/**
+ * This class tests the behavior of the Hybrid Subscriber Persistence Manager class.
+ */
+@WithCarbonHome
+@WithRegistry(injectToSingletons = {EntitlementServiceComponent.class})
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(files = {"dbscripts/h2.sql"})
+public class HybridSubscriberPersistenceManagerTest extends SubscriberPersistenceManagerTest {
+
+    JDBCSubscriberPersistenceManager jdbcSubscriberPersistenceManager;
+    RegistrySubscriberPersistenceManager registrySubscriberPersistenceManager;
+
+    public SubscriberPersistenceManager createSubscriberPersistenceManager() {
+
+        jdbcSubscriberPersistenceManager = new JDBCSubscriberPersistenceManager();
+        registrySubscriberPersistenceManager = new RegistrySubscriberPersistenceManager();
+        return new HybridSubscriberPersistenceManager();
+    }
+
+    @Test(priority = 5)
+    public void testGetSubscriberInDatabase() throws Exception {
+
+        jdbcSubscriberPersistenceManager.addSubscriber(sampleHolder1);
+
+        PublisherDataHolder subscriberFromDb =
+                subscriberPersistenceManager.getSubscriber(SAMPLE_SUBSCRIBER_ID_1, false);
+        assertEquals(subscriberFromDb.getPropertyDTO(SUBSCRIBER_ID_KEY).getValue(),
+                sampleHolder1.getPropertyDTO(SUBSCRIBER_ID_KEY).getValue());
+        assertEquals(subscriberFromDb.getPropertyDTO(SUBSCRIBER_URL_KEY).getValue(),
+                sampleHolder1.getPropertyDTO(SUBSCRIBER_URL_KEY).getValue());
+        assertEquals(subscriberFromDb.getPropertyDTO(SUBSCRIBER_USERNAME_KEY).getValue(),
+                sampleHolder1.getPropertyDTO(SUBSCRIBER_USERNAME_KEY).getValue());
+        assertEquals(subscriberFromDb.getPropertyDTO(SUBSCRIBER_PASSWORD_KEY).getValue(),
+                SAMPLE_ENCRYPTED_PASSWORD1);
+
+        // Retrieve the subscriber with the decrypted secrets.
+        PublisherDataHolder decryptedSubscriberFromStorage =
+                subscriberPersistenceManager.getSubscriber(SAMPLE_SUBSCRIBER_ID_1, true);
+        assertEquals(decryptedSubscriberFromStorage.getPropertyDTO(SUBSCRIBER_PASSWORD_KEY).getValue(),
+                SAMPLE_SUBSCRIBER_PASSWORD_1);
```

Comment:
This part is duplicated in two tests `testGetSubscriberInDatabase()` and `testGetSubscriberInRegistry()`, right? Shall we extract this to a method then?

Similar behavior is there in test methods as well.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/compare/3c296c23619d23b55ab1d555c12a582d7e16c508..762117e1906d5e7d3f44439db38b0b99dcb3315e